Apply global media settings except on product page

martinamarien opened the pull request:
**Why are these changes introduced?**
This PR applies the media global settings to the sections/features of the theme (excluding product page and featured product).

I will address product page media in a separate PR. 

Fixes #1083 
Fixes some items in #928

**What approach did you take?**
The styling was being re-applied to many different features/classes, so I created a utility class that can be applied to all media items. 

This is similar to the approach taken for content-containers

Applies to: 
- [x] Video section
- [x] Collage video
- [x] Collage image
- [x] Cart notification media
- [x] Cart item media (on cart page) 
- [x] Image with text media 

**Other Considerations**
I updated the slideshow class section to use "Section" so that it can inherit the section spacing global settings. 

**Feedback requested**
Questions for Meli/Reviewers: 

Should these styles also apply to: 
- [x] Media on page templates
- [ ] ~Image media in Footer~
- [ ] Slideshow media (separate issue)

**Demo links**
- [Editor](https://os2-demo.myshopify.com/admin/themes/127063719958/editor)

**Checklist**
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] N/A Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

melissaperreault commented:
> Questions for Meli/Reviewers:
> Should these styles also apply to:

- Media on page templates: _**YES**_
- Image media in Footer: _**NO**_, I would keep it low for now on footer as there could be various layout and needs.
- Slideshow media: _**MAYBE**_, To discuss the details with @nicklepine and @danielvan 🙏 I will report back!

martinamarien commented on assets/base.css:
```diff
@@ -2519,7 +2519,33 @@ details-disclosure > details {
 }
 
-.content-container--full-width {
+.global-media-settings {
```
Great point Ken (and Tyler). 
Changing this here will conflict with my other media branch. I think I'll opt to make a fast follow once they've both been merged and update the naming to `settings-media` and `settings-content-containers`.

melissaperreault reviewed (commented):
Thank you Martina! 🎉 Looking good!

Some observations:

- [ ] Not sure if this is relevant but since we are making the Collage blocks inherit the `Cards` settings from https://github.com/Shopify/dawn/pull/1090, this won't be necessary anymore for Collage `Video` ad `Image` block.
- [ ] Omit shadow settings, for `Cart template` and `Cart notification` thumbnails, like we'll do for `Product template` thumbnails.
- [ ] I wonder if we want to inherit `Media` settings on the [Collapsible content section](https://screenshot.click/22-01-bet2j-7f200.png) image until we apply `Content containers` to it from https://github.com/Shopify/dawn/issues/1043
- [ ] Not sure what we can do, when you link a page under the `Collapsible row block`, [the shadow gets cut](https://screenshot.click/22-01-494w3-4aj5q.png).

melissaperreault commented on assets/base.css:
```diff
@@ -2519,7 +2516,30 @@ details-disclosure > details {
+}
+
+.global-media-settings img,
+.global-media-settings iframe {
```
No, we don't want to apply the settings for inside the modal.

melissaperreault commented on sections/slideshow.liquid:
```diff
@@ -145,7 +145,7 @@
   "name": "t:sections.slideshow.name",
   "tag": "section",
-  "class": "spaced-section spaced-section--full-width",
+  "class": "section",
```
This PR https://github.com/Shopify/dawn/pull/1110 is taking care of adding padding setting to header and footer to space out the next section below it in relationship with other settings combo.

martinamarien commented:
@kmeleta @LucasLacerdaUX @melissaperreault 
I have addressed your feedback and rebased the branch. This means that the collage section should now be using the content containers global settings. 

Meli - these two points are now fixed.  
- [x] Not sure if this is relevant but since we are making the Collage blocks inherit the Cards settings from #1090, this won't be necessary anymore for Collage Video ad Image block.
- [x] Omit shadow settings, for Cart template and Cart notification thumbnails, like we'll do for Product template thumbnails.

I will address slideshow in a separate PR.

As for the collapsible content points. I will create an issue to tackle these seperately. 
- [ ] I wonder if we want to inherit Media settings on the Collapsible content section image until we apply Content containers to it from #1043
 - [ ] Not sure what we can do, when you link a page under the Collapsible row block, the shadow gets cut.

kmeleta reviewed (commented):
Just noticed one little code thing we can cleanup. Otherwise I think this is looking good to me

melissaperreault reviewed (approved):
Thank you Martina! 🎉 LGTM!
I made some comment below, we can ship and follow up on another PR once we have alignment on the path to move forward. 🙏 Thanks!

>  This means that the collage section should now be using the content containers global settings

- I think you meant `Cards` instead of `Content containers` global settings. 😄  Anyway, looks good!

- I also reviewed the issue from https://github.com/Shopify/dawn/issues/1099 about Image with text, looks good now 👍 

Other observations:

- [ ] I believe we might have no choice to do a systematic scale for the border for thumbnails in Cart notification and Cart template. cc. @danielvan @nicklepine 
   - Cart template
      - Mobile: the delete icon gets pushed when we increase the media border thickness. We would want to maintain the same positioning regardless of the range value. ([Video recording](https://screenshot.click/22-01-co6u0-531km.mp4))
      - Desktop: the thumbnails [are getting way too small](https://screenshot.click/22-01-dcxum-32ung.png)
   - Cart notification: [thumbnail gets small](https://screenshot.click/22-01-xnbwg-kyhvu.png) for an intent of large border on larger components.

nicklepine commented:
Thanks @melissaperreault . I created https://github.com/Shopify/dawn/issues/1164 to explore the systematic border thickness adjustment for these areas. This can be a Sev 2

LucasLacerdaUX commented on assets/component-deferred-media.css:
```diff
@@ -6,6 +6,12 @@
+  overflow: hidden;
+}
+
+.global-media-settings--full-width .deferred-media__poster {
+  border-radius: 0;
```
Now that `Collage` cards are using settings from `Cards`, the `Media` is still causing changes to the image radius in these blocks:
![image](https://user-images.githubusercontent.com/37168033/149174185-d329a6be-3a91-4819-8e9f-60feeefe4a2d.png)

Should this be replaced with:
```suggestion
.collage-card .deferred-media__poster {
  border-radius: 0;
```
or is this snippet accounting for something else? There's a `.collage-card .deferred-media__poster` selector in `collage.css` already if you prefer to add it there.

martinamarien replied:
^ This was actually still an issue on 1124, so I did fix it there.

LucasLacerdaUX commented on assets/component-image-with-text.css:
```diff
@@ -91,56 +91,58 @@
 @media screen and (max-width: 749px) {
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item:first-child:after,
-  .image-with-text.collapse-corners:not(.image-with-text--overlap) .grid__item .image-with-text__media {
+  .image-with-text.collapse-corners:not(.image-with-text--overlap) .image-with-text__media-item:after,
```
Setting the `Border width` to `0` makes the border radius look broken on both Desktop and Mobile.
<img src="https://user-images.githubusercontent.com/37168033/149451176-5ae492b2-7d2c-445e-8b78-174176b86da6.png" width="160"> <img src="https://user-images.githubusercontent.com/37168033/149451510-db09bfa5-46d3-43a1-91a5-80aa508c92d9.png" width="300">

Here's what it looks like if I increase `Border width` to `1`:

<img src="https://user-images.githubusercontent.com/37168033/149451289-5e017f6e-b40c-41fa-b111-e3f71caaf9ac.png" width="160"> <img src="https://user-images.githubusercontent.com/37168033/149451483-b159d468-6123-469a-8720-c17c46d87c99.png" width="300">

martinamarien replied:
We do need these styles applied, but I think we might need to revise the `collapse-corners` condition. 
That said, this is already something I can replicate on `main`, so I suggest we create an issue for this, and tackle in a separate PR

<img width="960" alt="Screen Shot 2022-01-14 at 10 22 46 AM" src="https://user-images.githubusercontent.com/26778637/149539936-73916c94-a9da-4399-842e-d63fe294c37c.png">

kmeleta replied:
I think that's actually a case we wanted, though Meli and I were going to discuss it more. @melissaperreault this our weird condition right?

LucasLacerdaUX reviewed (commented):
The issue I mentioned above is now fixed. I think there's still a bug with the `Image with Text` section.

LucasLacerdaUX reviewed (dismissed):
🐶👍
